### PR TITLE
Update APIs

### PR DESCRIPTION
**Pull request by plorenz**

### CODE REVIEW

File: channel.go [hunk 1]
```diff
@@ -39,23 +39,79 @@ type Channel interface {
...
+	SendListener() SendListener
+
+	// ReplyReceiver returns the ReplyReceiver to be invoked when a reply for the message or received, or nil if
+	// no ReplyReceiver should be invoked if or when a reply is received
+	ReplyReceiver() ReplyReceiver
 }
 
-type TimeoutSendContext interface {
-	SendContext
+// Envelope allow setting message priority and context. Message is an Envelope (as well as a Sendable)
```
Comment (andrewpmartinez) [inline channel.go:71]: `allow` to `allowS`

File: latency/latency.go [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+
+package latency
+
+import (
+	"github.com/michaelquigley/pfxlog"
+	"github.com/openziti/channel"
+	"github.com/openziti/foundation/channel2"
```
Comment (andrewpmartinez) [inline latency/latency.go:22]: channel2? I assume because of the routers?

Comment (plorenz) [reply latency/latency.go:22]: no, just missed it. Fixing.

File: channel.go [hunk 3]
```diff
@@ -39,23 +39,79 @@ type Channel interface {
 	GetTimeSinceLastRead() time.Duration
 }
 
-type SendContext interface {
+type Sender interface {
```
Comment (andrewpmartinez) [inline channel.go:42]: I like this name better - both idiomatic and not using the work 'context'.

**Review by andrewpmartinez — APPROVED** [review]

File: channel.go (hunk 1 above)
Comment (plorenz) [reply channel.go:71]: fixed

File: latency/latency.go (hunk 2 above)
Comment (plorenz) [reply latency/latency.go:22]: fixed

**Pull request merged**